tests/integration/test_http_client.py

```diff
@@ -2,18 +2,25 @@
 import pytest
 
 import solana.system_program as sp
-from solana.rpc.api import DataSliceOpt
+from solana.rpc.api import DataSliceOpt, AsyncClient
 from solana.transaction import Transaction
 
-from .utils import assert_valid_response, confirm_transaction
+from .utils import assert_valid_response, confirm_transaction, aconfirm_transaction, compare_responses_without_ids
 
 
 @pytest.mark.integration
-def test_request_air_drop(stubbed_sender, test_http_client):
+def test_request_air_drop(stubbed_sender, test_http_clients):
```

Comment:
Can we test the async client separate from the old client? Having both synchronous and async tests in the same method makes maintenance more difficult.

Reply:
This is possible, but we'll have a lot of duplicated code without a big refactor of the test suite. Plus there are places where we just test that the response the async client received is the same as that of the sync client, and we'll lose that (without more refactoring).

To get this over the line I'll just duplicate all the old pre-async tests and make async versions of them. Your call if that's what you want

Reply:
Yes. Duplicate tests are fine, it's just less confusing. We can decide to refactor and try to reduce duplicates later.